### PR TITLE
Adding System-as-root related info for H9 build 

### PR DESCRIPTION
**Pull request by mathew-dennis**
adding content related to system-as-root .

### CODE REVIEW

**Review by abkro — COMMENTED** [review]
Small spelling mistake:
to be build as > to be built as

**Comment by doniks** [comment]
@mathew-dennis are you able to follow up the review?

**Comment by mathew-dennis** [comment]
> @mathew-dennis are you able to follow up the review?

sorry I missed @abkro 's review, added the change,please review now
and thanks @abkro

File: porting/build_and_boot/H9_build.rst [hunk 1]
```diff
@@ -63,6 +63,28 @@ If your makefile already includes a line beginning with ``BOARD_KERNEL_CMDLINE``
 
 .. Note::
     In rare cases the bootloader overwrites the kernel command line argument, rendering the setting above useless. This is the case for the Google Pixel 3a (sargo). To deal with this issue, replicate `this commit <https://github.com/fredldotme/android_kernel_google_bonito/commit/d0741dded3907f2cf4ecdc02bfcb74fc252763ff>`_. 
+    
+Setting system as root
+^^^^^^^^^^^^^^^^^^^^^^
+
+For halium 9, we need the system image to be built as system-as-root (for treble devices it should already be there ). Please check your ''boardconfig.mk'' for the following line, if it's there you can skip this step::
```
Comment (doniks) [inline porting/build_and_boot/H9_build.rst:70]: halium 9 -> Halium 9

"there )" -> "there)" remove extra space

formatting of boardconfig should be with two backticks not two single quotes, ie ``boardconfig`` https://sphinx-tutorial.readthedocs.io/step-1/#code-samples

**Review by doniks — CHANGES_REQUESTED** [review]

**Comment by doniks** [comment]
hey thanks for the quick followup. however, also, sorry, I found a few more points pls check

File: porting/build_and_boot/H9_build.rst [hunk 2]
```diff
@@ -63,6 +63,28 @@ If your makefile already includes a line beginning with ``BOARD_KERNEL_CMDLINE``
...
+    BOARD_BUILD_SYSTEM_ROOT_IMAGE := true
+
+If the above is not there, we have to add it.
+
+As this change makes the android root read-only, we may have to ship mount points for some important partitions like '' firmware'' ,'' persist'' with the system image, for this we can use the line::
+
+    BOARD_ROOT_EXTRA_FOLDERS :=
+     /firmware
+     /dsp
+     /persist
```
[the remark this answers is not in the document]
Comment (mathew-dennis) [reply porting/build_and_boot/H9_build.rst:81]: backslashes are needed or we may have to repeat the first line 3 times.

**Comment by mathew-dennis** [comment]
@doniks done ;-) 

also, could you please check with TheKit before merging, coz when I made this pr ,there was a comment in the porting group that this content is no longer relevant.thanks

**Comment by doniks** [comment]
> @doniks done ;-)

Awesome. Thanks. 

> 
> also, could you please check with TheKit before merging, coz when I made this pr ,there was a comment in the porting group that this content is no longer relevant.thanks

Of course! :D @NotKit could you weigh in here?

**Review by doniks — APPROVED** [review]

**Comment by doniks** [comment]
Mhm. We've been sitting on this for a while now. No feedback from notkit, but I also think this is not a very risky change. Let's merge and fix it later if needed

**Pull request merged**